packages/components/src/templates/next/components/internal/SearchableTable/SearchableTable.tsx

```diff
@@ -0,0 +1,198 @@
+"use client"
+
+import { useId, useRef, useState } from "react"
+
+import type { SearchableTableProps } from "~/interfaces"
+import { tv } from "~/lib/tv"
+import { PaginationControls } from "../PaginationControls"
+import { SearchField } from "../Search"
+import { MAX_NUMBER_OF_COLUMNS, PAGINATION_MAX_ITEMS } from "./constants"
+import { getFilteredItems } from "./getFilteredItems"
+import { getPaginatedItems } from "./getPaginatedItems"
+
+const createSearchableTableStyles = tv({
+  slots: {
+    container: "mx-auto w-full",
+    title: "prose-display-md break-words text-base-content-strong",
+    search: "mt-9",
+    tableContainer: "mt-7 overflow-x-auto",
+    table: "w-full border-collapse border-spacing-0",
+    tableRow: "text-left",
+    tableCell:
+      "max-w-40 break-words border border-base-divider-medium px-4 py-3 align-top last:max-w-full [&_li]:my-0 [&_li]:pl-1 [&_ol]:mt-0 [&_ol]:ps-5 [&_ul]:mt-0 [&_ul]:ps-5",
+    emptyState:
+      "flex flex-col items-center justify-center gap-8 self-stretch px-10 py-20 pt-24",
+    emptyStateHeadings: "text-center",
+    emptyStateTitle: "prose-title-lg-regular",
+    emptyStateSubtitle:
+      "text-base-content-default prose-headline-lg-regular mt-3",
+    emptyStateButton:
+      "prose-headline-base-medium text-link hover:text-link-hover",
+    pagination: "mt-8 flex w-full justify-center lg:justify-end",
+  },
+  variants: {
+    isHeader: {
+      true: {
+        tableCell:
+          "bg-base-canvas-backdrop [&_ol]:prose-label-md-medium [&_p]:prose-label-md-medium",
+      },
+      false: {
+        tableCell:
+          "bg-base-canvas-alt [&_ol]:prose-body-sm [&_p]:prose-body-sm",
+      },
+    },
+    bold: {
+      true: {
+        emptyStateTitle: "text-base-content-strong",
+      },
+      false: {
+        emptyStateTitle: "text-base-content-subtle",
+      },
+    },
+  },
+})
+
+const compoundStyles = createSearchableTableStyles()
+
+const SearchableTable = ({ title, headers, items }: SearchableTableProps) => {
+  const [search, setSearch] = useState("")
+  const [currPage, setCurrPage] = useState(1)
+  const titleId = useId()
+
+  const maxNoOfColumns = Math.min(
+    headers.length,
+    ...items.map((row) => row.length),
+    MAX_NUMBER_OF_COLUMNS,
+  )
+  const filteredItems = getFilteredItems({
+    items,
+    searchValue: search,
+  })
+  const paginatedItems = getPaginatedItems({
+    items: filteredItems,
+    currPage,
+    itemsPerPage: PAGINATION_MAX_ITEMS,
+  })
+
+  const isInitiallyEmpty = items.length === 0 || maxNoOfColumns === 0
+  const isFilteredEmpty = items.length !== 0 && filteredItems.length === 0
+
+  const sectionTopRef = useRef<HTMLDivElement>(null)
+  const onPageChange = () => {
+    sectionTopRef.current?.scrollIntoView({
+      block: "start",
+      behavior: "smooth",
```

Comment:
nit(a11y): might want to change the behaviour depending on user's `prefers-reduced-motion` setting; should be instant if true.

Can probably use a [hook](https://github.com/anatoliygatt/use-prefers-reduced-motion/blob/master/src/usePrefersReducedMotion.ts) to retrieve that value

[Further reading ](https://www.joshwcomeau.com/react/prefers-reduced-motion/)(or use his [hook](https://www.joshwcomeau.com/snippets/react-hooks/use-prefers-reduced-motion/) if you want, but feels old lol)

Reply:
Removed this, want to standardise this with the behaviour currently in the collections layout.